File: backend/src/instamart_engine/api/schemas/insights.py
"""Response DTOs for the (minimal, read-only) insights surface — mirrors
`api/schemas/themes.py`'s exact shape: list + detail, nothing more.
Phase 4's insights service/repository existed with no API surface at all
until now (audit-2026-07-31.md F-13/R-7)."""

from uuid import UUID

from pydantic import BaseModel


class InsightSummaryResponse(BaseModel):
    id: UUID
    insight_type: str
    title: str
    finding: str
    confidence_level: str
    confidence_score: float | None
    opportunity_score: float | None


class InsightEvidencePreviewResponse(BaseModel):
    id: UUID
    excerpt: str
    evidence_role: str
    source_connector_key: str
    published_at: str | None


class InsightThemeLinkResponse(BaseModel):
    theme_id: UUID
    theme_name: str
    relationship_type: str


class InsightDetailResponse(InsightSummaryResponse):
    interpretation: str
    affected_context: str | None
    product_implication: str | None
    validation_recommendation: str | None
    insight_set_id: UUID
    themes: list[InsightThemeLinkResponse]
    evidence: list[InsightEvidencePreviewResponse]


class InsightListResponse(BaseModel):
    insight_set_id: UUID | None
    insight_set_status: str | None
    theme_set_id: UUID | None
    analysis_run_id: UUID | None
    insights: list[InsightSummaryResponse]

File: frontend/src/lib/api/insights.ts
import { useQuery } from "@tanstack/react-query";

import { apiClient } from "@/lib/api/client";

export interface InsightSummary {
  id: string;
  insight_type: string;
  title: string;
  finding: string;
  confidence_level: string;
  confidence_score: number | null;
  opportunity_score: number | null;
}

export interface InsightEvidencePreview {
  id: string;
  excerpt: string;
  evidence_role: string;
  source_connector_key: string;
  published_at: string | null;
}

export interface InsightThemeLink {
  theme_id: string;
  theme_name: string;
  relationship_type: string;
}

export interface InsightDetail extends InsightSummary {
  interpretation: string;
  affected_context: string | null;
  product_implication: string | null;
  validation_recommendation: string | null;
  insight_set_id: string;
  themes: InsightThemeLink[];
  evidence: InsightEvidencePreview[];
}

export interface InsightList {
  insight_set_id: string | null;
  insight_set_status: string | null;
  theme_set_id: string | null;
  analysis_run_id: string | null;
  insights: InsightSummary[];
}

export function useInsights() {
  return useQuery({
    queryKey: ["insights"],
    queryFn: () => apiClient.get<InsightList>("/api/v1/insights"),
  });
}

export function useInsight(insightId: string | null) {
  return useQuery({
    queryKey: ["insight", insightId],
    queryFn: () => apiClient.get<InsightDetail>(`/api/v1/insights/${insightId}`),
    enabled: Boolean(insightId),
  });
}
